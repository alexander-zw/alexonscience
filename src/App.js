import DocumentMeta from 'react-document-meta';
import './App.css';
import hello_there from './images/hello-there.png';
import general_kenobi from './images/general-kenobi.png';

const meta = {
    title: 'ALEX on Science',
    description: 'The official ALEX on Science website',
    canonical: 'https://alexonscience.com',
    image: hello_there,
    meta: {
        charset: 'utf-8',
        name: {
            keywords: 'alex,alexander,wu,science,youtube'
        }
    }
}

function App() {
    return (
        <div>
            <DocumentMeta {...meta} />

            <div className="title-div">
                <span className="title">Alexander Wu</span>
            </div>

            <div className="text-div">
                <p>Psst... check out my <a href="https://tinyurl.com/alexonscience">Youtube Channel</a>!</p>
            </div>

            <div id="kenobi-image">
                <img className="bottom" src={general_kenobi} alt="Hello there"/>
                <img className="top" src={hello_there} alt="General Kenobi"/>
            </div>
        </div>
    );
}

export default App;
