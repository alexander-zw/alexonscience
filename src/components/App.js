/**
 * The main component of the site, including the navigation bar, main content,
 * and footer.
 *
 * TODO:
 * Add portfolio
 * Add projects
 * Improve homepage
 * Add blog
 */
import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import Error from './Error';
import Navigation, { navigationViews } from './Navigation';
import Footer from './Footer';

function App() {
    const routesComponent = navigationViews.map((route, index) => (
        <Route path={route.path} component={route.component} exact={route.exact} key={index}/>
    ));

    return (
        <BrowserRouter>
            <div className="non-footer-content">
                <Navigation />
                <Switch>
                    {routesComponent}
                    <Route component={Error}/>
                </Switch>
            </div>
            <Footer />
        </BrowserRouter>
    );
}

export default App;