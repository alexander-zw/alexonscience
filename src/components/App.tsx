import React, { useEffect } from "react";
import { BrowserRouter, Route, Switch, useLocation } from "react-router-dom";

import science_banner from "../images/science_banner.jpg";
import Error404 from "./Error404";
import Footer from "./Footer";
import Navigation from "./Navigation";
import { routableViews } from "./subcomponents/AllViews";

/**
 * The main component of the site, including the navigation bar, main content,
 * and footer.
 *
 * TODO:
 * Add blog
 * Fix resume for mobile
 * Mobile friendly nav bar
 * Prerendering
 */

// Scroll to the top of page whenever user switches views.
function ScrollToTop() {
    const { pathname } = useLocation();

    useEffect(() => {
        window.scrollTo(0, 0);
    }, [pathname]);

    return null;
}

function App() {
    const routesComponent = Object.entries(routableViews).map(([path, route]) => (
        <Route path={path} component={route.component} exact={route.exact} key={path} />
    ));

    return (
        <BrowserRouter>
            <ScrollToTop />
            <div className="non-footer-content">
                <div>
                    <img id="science-banner-image" src={science_banner} alt="ALEX on Science" />
                </div>
                <Navigation />
                <Switch>
                    {routesComponent}
                    <Route component={Error404} />
                </Switch>
            </div>
            <Footer />
        </BrowserRouter>
    );
}

export default App;
