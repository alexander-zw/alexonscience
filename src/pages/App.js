import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import Home from './Home';
import Art from './Art';
import Error from './Error';
import Navigation from './Navigation';

class App extends Component {
  render() {
    return (
        <BrowserRouter>
            <div>
                <Navigation />
                <Switch>
                    <Route path="/" component={Home} exact/>
                    <Route path="/art" component={Art}/>
                    <Route component={Error}/>
                </Switch>
            </div> 
        </BrowserRouter>
    );
  }
}

export default App;