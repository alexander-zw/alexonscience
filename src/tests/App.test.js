/**
 * This test checks the whole page, so it is able to test the navigation bar
 * (Navigation.js) and footer (Footer.js).
 */
import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import App from '../components/App';

it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});
