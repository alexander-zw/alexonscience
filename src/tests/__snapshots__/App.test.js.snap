// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="non-footer-content"
  >
    <div>
      <img
        alt="ALEX on Science"
        id="science-banner-image"
        src="science_banner.jpg"
      />
    </div>
    <div
      className="nav-bar text-div"
    >
      <div
        className="nav-option-div"
      >
        <a
          aria-current="page"
          className="nav-option active-nav-option"
          href="/"
          onClick={[Function]}
          onUpdate={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </div>
      <div
        className="nav-option-div"
      >
        <a
          aria-current={null}
          className="nav-option"
          href="/youtube"
          onClick={[Function]}
          onUpdate={[Function]}
        >
          YouTube
        </a>
      </div>
      <div
        className="nav-option-div"
      >
        <a
          aria-current={null}
          className="nav-option"
          href="/resume"
          onClick={[Function]}
          onUpdate={[Function]}
        >
          Resume
        </a>
      </div>
      <div
        className="nav-option-div"
      >
        <a
          aria-current={null}
          className="nav-option"
          href="/projects"
          onClick={[Function]}
          onUpdate={[Function]}
        >
          Projects
        </a>
      </div>
      <div
        className="nav-option-div"
      >
        <a
          aria-current={null}
          className="nav-option"
          href="/art"
          onClick={[Function]}
          onUpdate={[Function]}
        >
          Art
        </a>
      </div>
      <div
        className="nav-option-div"
      >
        <a
          aria-current={null}
          className="nav-option"
          href="/contact"
          onClick={[Function]}
          onUpdate={[Function]}
        >
          Contact Me
        </a>
      </div>
    </div>
    <div>
      <div
        className="text-div bottom-margin"
      >
        <p>
          Hello there! My name is Alexander Wu. I am a graduate student who enjoys maintaining a YouTube channel and working on personal coding projects. Feel free to explore around!
        </p>
        <p>
          Check out some of my awesome 
          <a
            href="/projects"
            onClick={[Function]}
          >
            projects
          </a>
          ! Or, if you
          '
          d like to learn about me (I
          '
          m flattered), take a look at my
           
          <a
            href="/resume"
            onClick={[Function]}
          >
            resume
          </a>
          .
        </p>
        <span
          className="videos-title"
        >
          Recent Uploads
        </span>
        <div
          className="embedded-video-container"
        >
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen={true}
            className="embedded-video"
            frameBorder="0"
            height={200}
            src="https://www.youtube.com/embed/cHAVhoZx2os"
            title="Most recent video"
            width={355.55555555555554}
          />
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen={true}
            className="embedded-video"
            frameBorder="0"
            height={200}
            src="https://www.youtube.com/embed/cI5lkif-V1c"
            title="Most recent video"
            width={355.55555555555554}
          />
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen={true}
            className="embedded-video"
            frameBorder="0"
            height={200}
            src="https://www.youtube.com/embed/wuX_QyS0C3A"
            title="Most recent video"
            width={355.55555555555554}
          />
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen={true}
            className="embedded-video"
            frameBorder="0"
            height={200}
            src="https://www.youtube.com/embed/Vsgek3M1PZw"
            title="Most recent video"
            width={355.55555555555554}
          />
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen={true}
            className="embedded-video"
            frameBorder="0"
            height={200}
            src="https://www.youtube.com/embed/Jb6nFKPov-w"
            title="Most recent video"
            width={355.55555555555554}
          />
        </div>
        <div
          className="MuiPaper-root MuiAccordion-root expansion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="MuiButtonBase-root MuiAccordionSummary-root"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiAccordionSummary-content"
            >
              What are those drawings in your banner?
            </div>
            <div
              aria-disabled={false}
              aria-hidden={true}
              className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role={null}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="MuiCollapse-container MuiCollapse-hidden"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                >
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    As you might be able to tell, I love drawing, even if I'm not that good at it. Here you can see a Turing machine, AlphaGo, Albert Einstein, some DNA, and Rosalind Franklin. Franklin is particularly underappreciated - she was not recognized for a Nobel prize only because she had passed away by the time it was awarded.
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="text-div"
    id="footer-div"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
    <div
      className="left-half"
    >
      Alexander Wu
      <br />
      Let there be light.
    </div>
    <div
      className="right-half"
    >
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact me
      </a>
    </div>
  </footer>,
]
`;
